src/reducers-actions/orders.ts

```diff
@@ -38,21 +79,67 @@ export const reducer = (state: OrdersState, action: ReducerActionType): OrdersSt
         payload: { orders },
       } = action
 
-      // default sorting order is ascending by order id
-      // since we are starting from 0, reverse to have the latest on top
-      // make a copy first (slice(0)) to avoid mutating current state
-      const reversedOrders = orders.slice(0).reverse()
+      const reversedOrders = orders
+        // make a copy first (slice(0)) to avoid mutating current state
+        .slice(0)
+        // default sorting order is ascending by order id
+        // since we are starting from 0, reverse to have the latest on top
+        .reverse()
+        // remove deleted orders
+        .filter(order => !isOrderDeleted(order))
 
       return { ...state, orders: reversedOrders }
     }
+    case 'UPDATE_ORDERS': {
+      const {
+        payload: { orders: newOrders },
+      } = action
+      const { orders: currentOrders } = state
+
+      // Classify new orders between existing and deleted,
+      // keeping track of the ids and indexes
+      const { activeOrders, deletedOrdersIds } = classifyOrders(newOrders)
+
+      // Filter deleted orders and update existing orders
+      const filteredAndUpdatedOrders = currentOrders.reduce((acc: AuctionElement[], order) => {
+        if (activeOrders[order.id] !== undefined) {
+          // It's on activeOrders map? it has been updated.
+          // Get the updated from newOrders list by the index in the activeOrders map
+          acc.push(newOrders[activeOrders[order.id]])
+          // Remove from activeOrders map so we know it's not a new order
+          delete activeOrders[order.id]
```

Comment:
would be preferable as a `Set` with `set.delete(order.id)`

Reply:
But in that case, I wouldn't have all the info I need. Having only the order.id in the set means I'd have to search `newOrders` array for every order found.

I can give it a try using a Map for a better interface.

Reply:
No longer in use with Dima's suggestion